### ISSUE TITLE
Sample to create Duplicate Rule using Metadata API

### ISSUE THREAD

**Issue by SFFreelancer** [financialforcedev/apex-mdapi#168]
Hello, 

I am trying to create DuplicateRule and reorder them using MetadataService Class, but I did not find luck can anybody share code snippets or guidance.

Thanks & Regards,
Shrikant
srbagal.ceo@gmail.com

**Comment by vamsikatragadda** [comment]
I have the same issue with code:
 MetadataService.MetadataPort service = new MetadataService.MetadataPort();
        MetadataService.DuplicateRule customObject = new MetadataService.DuplicateRule();
        customObject.actionOnInsert='allow';
        customObject.actionOnUpdate='allow';
        customObject.isActive=true;
        customObject.masterLabel='Main Label';
        customObject.sortOrder=3;

MetadataService.SaveResult [] results = service.createMetadata(new MetadataService.Metadata[] { customObject });

Error: Line: 10, Column: 97
Initial expression is of incorrect type, expected: MetadataService.Metadata

**Comment by afawcett** [comment]
Yes it looks like DuplicateRule is not extending the Metadata base class. You can make the changes manually, just follow the pattern of one of the other types like CustomField for an example. Meanwhile i'll flag this as a bug.

**Comment by md6759** [comment]
Hello Andrew,

I unsubscribed but I'm still receiving emails.

Thank you.

**Comment by afawcett** [comment]
@md6759 i am not sure how to help? I do not control GitHub... Sorry.
